### PR TITLE
sql/ocsql: allow configuration of ocsql instrumentation

### PR DESCRIPTION
**Pull request by ijt**

### CODE REVIEW

**Comment by ijt** [comment]
Fixes #416

File: mysql/cloudmysql/cloudmysql.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type Params struct {
 }
 
 // Open opens a Cloud SQL database.
-func Open(ctx context.Context, certSource proxy.CertSource, params *Params) (*sql.DB, error) {
+func Open(ctx context.Context, certSource proxy.CertSource, params *Params, traceOpts ocsql.TraceOption) (*sql.DB, error) {
```
Comment (vangent) [inline mysql/cloudmysql/cloudmysql.go]: It's unfortunate that this argument list is getting kind of long, and that this is a breaking change. Maybe add to Params instead?

File: pubsub/pubsub.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
+// Copyright 2018 The Go Cloud Authors
```
Comment (vangent) [inline pubsub/pubsub.go]: This file doesn't look like it belongs in this PR.

File: mysql/cloudmysql/cloudmysql.go (hunk 1 above)
Comment (vangent) [inline mysql/cloudmysql/cloudmysql.go]: It looks like ocsql.TraceOption is a func; what you have here only allows a single one. I.e., what if I want WithRowsAffected and WithRowsClosed?

Maybe this should be a *ocsql.TraceOptions instead?

File: mysql/cloudmysql/cloudmysql.go [hunk 3]
```diff
@@ -76,21 +76,26 @@ func Open(ctx context.Context, certSource proxy.CertSource, params *Params) (*sq
 		Passwd:               params.Password,
 		DBName:               params.Database,
 	}
-	return sql.OpenDB(connector(cfg.FormatDSN())), nil
+	var c connector
+	c.dsn = cfg.FormatDSN()
+	c.dbDriver = ocsql.Wrap(mysql.MySQLDriver{}, traceOpts)
```
Comment (vangent) [inline mysql/cloudmysql/cloudmysql.go]: Does this work if traceOpts is nil?

File: mysql/cloudmysql/cloudmysql.go (hunk 1 above)
Comment (ijt) [reply mysql/cloudmysql/cloudmysql.go]: Done.

Comment (ijt) [reply mysql/cloudmysql/cloudmysql.go]: Done.

File: pubsub/pubsub.go (hunk 2 above)
Comment (ijt) [reply pubsub/pubsub.go]: Done.

File: mysql/cloudmysql/cloudmysql.go (hunk 1 above)
Comment (ijt) [reply mysql/cloudmysql/cloudmysql.go]: Sorry, I misunderstood how these are supposed to be used. An example is here:

https://github.com/opencensus-integrations/ocsql/blob/f67f654fe0de226036abcecd6354c4b8205b1afb/driver.go#L81

Comment (ijt) [reply mysql/cloudmysql/cloudmysql.go]: Fixed it by making it varargs like it's supposed to be.

File: mysql/cloudmysql/cloudmysql.go (hunk 3 above)
Comment (ijt) [reply mysql/cloudmysql/cloudmysql.go]: Fixed with varargs.

**Pull request merged**